dotfiles/run.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+DOTFILES_URI="${DOTFILES_URI}"
+DOTFILES_USER="${DOTFILES_USER}"
+
+if [ -n "$${DOTFILES_URI// }" ]; then
+  if [ -z "$DOTFILES_USER" ]; then
+    DOTFILES_USER="$USER"
+  fi
+
+  echo "✨ Applying dotfiles for user $DOTFILES_USER"
+
+  if [ "$DOTFILES_USER" = "$USER" ]; then
+    coder dotfiles "$DOTFILES_URI" -y 2>&1 | tee ~/.dotfiles.log
+  else
+    # The "eval echo ~$DOTFILES_USER" part is used to dynamically get the home directory of the user, see https://superuser.com/a/484280
+    # sudo -u coder sh -c 'eval echo ~$DOTFILES_USER' -> "/home/coder"
+    # sudo sh -c 'eval echo ~$DOTFILES_USER'          -> "/root"
```

Comment:
Why do these look different? Both can just run as the normal user, no sudo involved. The variable name being the same is confusing.

```suggestion
    # eval echo ~$USER'                   -> "/home/coder"
    # eval echo ~${DOTFILES_USER:-root}'  -> "/root"
```

Reply:
I changed the variable the other day because I got mixed up but the examples were actually meant to use `$USER` and since we use single quotes that command would've been run as the current user within the sudo command, so `$USER` was correct in the previous commits.

Now the way we resolve the home changed though so we can indeed change it.

Reply:
6ffc79f

Reply:
Actually, this was causing linting issues so I changed it to the following, which is clearer anyways in my opinion.

```bash
  # The `eval echo ~"$DOTFILES_USER"` part is used to dynamically get the home directory of the user, see https://superuser.com/a/484280
  # eval echo ~coder -> "/home/coder"
  # eval echo ~root  -> "/root"
``

0b8fe33